dev/NavigationView/NavigationView_rs1_themeresources.xaml

```diff
@@ -794,6 +794,12 @@
                         <Rectangle 
                             x:Name="PointerRectangle" 
                             Fill="Transparent" />
+                        <Border		
+                            x:Name="RevealBorder"		
+                            BorderBrush="{TemplateBinding BorderBrush}"		
+                            BorderThickness="{TemplateBinding BorderThickness}"
+                            contract4NotPresent:Visibility="Collapsed" >
```

Comment:
I thought about that too. This Border is also used in the "MUX_NavigationViewItemPresenterStyleWhenOnLeftPane" style . Should we remove the contract4NotPresent for that Border element too?

Reply:
@kaiguo what would you recommend?

Reply:
Yeah I think it's safe to remove them altogether.

@chingucoding You probably need to update reveal states and maybe border brushes too in the visual state setters. Take a look at "MUX_NavigationViewItemPresenterStyleWhenOnLeftPane"

Reply:
> @chingucoding You probably need to update reveal states and maybe border brushes too in the visual state setters. Take a look at "MUX_NavigationViewItemPresenterStyleWhenOnLeftPane"

I don't think this is necessary since the reveal works just as expected:

| Dark theme | Light theme |
|--------------|---------------|
| ![navtop-settingsbutton-reveal-dark](https://user-images.githubusercontent.com/16122379/64002935-84868d00-cb0b-11e9-8c73-9c2e5eb76955.gif)|![navtop-settingsbutton-reveal-light](https://user-images.githubusercontent.com/16122379/64002936-84868d00-cb0b-11e9-93f9-cd29d07b4f72.gif)|

Reply:
![ezgif-2-7690b84940d6](https://user-images.githubusercontent.com/4424330/64041050-4a20ee00-cb13-11e9-84d4-004410dbf533.gif)

This is very subtle but it's not only the approaching animation, there are also hover and pressed animations, which requires setting reveal brush states.

https://docs.microsoft.com/en-us/windows/uwp/design/style/reveal#using-the-control-template-to-add-reveal

Reply:
Should be fixed now. 
@kaiguo Thank you for the pointing out the right places to fix!

Reply:
@chingucoding thanks for the fix!

There's another related issue, you probably have noticed, the items on top nav don't have reveal either. Would you like to contribute a fix for that too 😊?

![20190831113543](https://user-images.githubusercontent.com/4424330/64067830-d7337800-cbe3-11e9-8fc8-33f919d666dd.png)

Reply:
@kaiguo Of course! Should the reveal be the same as for the settings button?